File: safety/snip/test_make_appender_dangle.rs
// Rust test_make_appender_dangle

fn test_make_appender_dangle() {
    let append34 = {
        let suffix = vec![3, 4];
        make_appender(&suffix) // Won't compile.
    };
    assert_eq!(vec![1, 2, 3, 4], append34(vec![1, 2]));
}

File: safety/snip/TestMakeAppenderDangle.go
func TestMakeAppenderDangle(t *testing.T) {
	append34 := func() func([]int) []int {
		suffix := []int{3, 4}
		return MakeAppender(suffix)
	}()
	want := []int{1, 2, 3, 4}
	if got := append34([]int{1, 2}); !reflect.DeepEqual(got, want) {
		t.Errorf("append34([]int{1, 2}) = %v; want %v", got, want)
	}
}
